src/useQuery/useQuery.ts

```diff
@@ -38,19 +50,23 @@ const useQuery: (query: Query, variables?: any) => any = (query, variables) => {
     (domain, action, options, updateQuery) => {
       const key = generateQueryCacheKey({ domain, action, options });
 
-      // Check for previous results
-      const cacheResult = selectQueryData(store.getState(), key);
-      if (cacheResult) {
-        setState({ loading: false, data: cacheResult, error: undefined });
-        return;
+      if (!ignoreCache) {
+        // Check for previous results
+        const cacheResult = selectQueryData(store.getState(), key);
+        if (cacheResult) {
+          setState({ loading: false, data: cacheResult, error: undefined });
+          return;
+        }
       }
 
       setState({ loading: true, error: undefined });
 
       // @TODO this promise should be cancellable
       API[domain][action](options)
         .then(data => {
-          store.dispatch(cacheQueryResult({ domain, action, options, data }));
+          if (!ignoreCache) {
```

Comment:
I'm not sure if this should be configured by the same option. I can imagine that you would still want to cache the result even if you didn't want to use the cache when fetching it. It's ok for now of course, but just something we can keep in the back of our mind.

Reply:
Good point, thought about it as well, going to take a look at doing it differently and see how that goes, because I'll have to write the test differently as well